contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java

```diff
@@ -58,10 +58,10 @@ private boolean filterSubstation(Substation substation) {
             return false;
         }
         Country injectionCountry = substation.getCountry().orElse(null);
-        if (injectionCountry == null) {
+        if (injectionCountry == null && !countries.isEmpty()) {
```

Comment:
From what I understand `countries` should not be `null`: add a `Objects.requireNonNull` in the constructor

Reply:
Done